src/QirRuntime/lib/QIR/qirTypes.hpp

```diff
@@ -66,18 +60,109 @@ struct QirArray
     ~QirArray();
 
     char* GetItemPointer(int64_t index);
-    QUBIT* GetQubit(int64_t index);
     void Append(const QirArray* other);
 };
 
-
 /*==============================================================================
     QirString is just a wrapper around std::string
 ==============================================================================*/
 struct QirString
 {
-    std::atomic<long> refCount;
+    long refCount = 1;
     std::string str;
 
     QirString(std::string&& str);
 };
+
+/*==============================================================================
+    The types and methods, expected by QIR for tuples:
+    %TupleHeader = type { i32, i32 }
+    declare %TupleHeader* @__quantum__rt__tuple_create(i64)
+
+    Argument passed to __quantum__rt__tuple_create is the size (in bytes) of the tuple, including the header.
+    For example:
+    ; to calculate the size of a tuple pretend having an array of them and get
+    ; offset to the first element
+    %t1 = getelementptr { %TupleHeader, %Callable*, %Array* }, { %TupleHeader, %Callable*, %Array* }* null, i32 1
+    ; convert the offset to int and call __quantum__rt__tuple_create
+    %t2 = ptrtoint { %TupleHeader, %Callable*, %Array* }* %t1 to i64
+    %0 = call %TupleHeader* @__quantum__rt__tuple_create(i62 %t2)
+
+    Notice, that the TupleHeader is included into the Tuple's buffer.
+==============================================================================*/
+struct QirTupleHeader
+{
+    int refCount = 0;
+    int size = sizeof(QirTupleHeader);

```

Comment:
Add a comment that the size field should be set to sizeof(header) + size_of_data (default initializing to the former might be confusing for the reader)